backend/database/versions/b2b6ffe65512_harmonize_schema_state_staging.py

```diff
@@ -0,0 +1,90 @@
+"""harmonize_schema_state_staging
+
+Revision ID: b2b6ffe65512
+Revises: 4ddb19dcc4e8
+Create Date: 2020-08-28 12:52:59.416278
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "b2b6ffe65512"
+down_revision = "fa0950f6df19"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_table("dataset_deployment")
+    op.alter_column("dataset", "project_id", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("dataset", "project_status", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("dataset", "revision", existing_type=sa.INTEGER(), nullable=True)
+    op.alter_column("deployment_directory", "created_at", existing_type=postgresql.TIMESTAMP(), nullable=False)
+    op.alter_column("deployment_directory", "dataset_id", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("deployment_directory", "updated_at", existing_type=postgresql.TIMESTAMP(), nullable=False)
+    op.create_foreign_key(None, "deployment_directory", "dataset", ["dataset_id"], ["id"])
+    op.alter_column("project", "needs_attestation", existing_type=sa.BOOLEAN(), nullable=True)
```

Comment:
Why is this nullable? Should just be a true or false and can default to false?

Reply:
Similar to above, it's just reading the ORM code here, which specifies this as `needs_attestation = Column(Boolean)` which means nullable and no default. But yes, your suggestion sounds sensible.